pkg/ipamd/ipamd.go

```diff
@@ -644,12 +651,109 @@ func (c *IPAMContext) updateLastNodeIPPoolAction() {
 	logPoolStats(total, used, c.maxIPsPerENI)
 }
 
-func (c *IPAMContext) tryAllocateENI() error {
+
+// For each ENIConfig we want to make sure there is at least
+// one matching interface attached
+func (c *IPAMContext) globalAllocateENI(enis []awsutils.ENIMetadata) error {
+
+	if c.eniConfig == nil {
+		return fmt.Errorf("eniConfig does not exist")
+	}
```

Comment:
When would this be nil? Doesn't it get set to `"default"` by default?

Reply:
I don't think there is a good reason for this to happen today, I just put in the test as an additional sanity check.